assets/js/base/context/hooks/shipping/use-shipping-data.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import { CART_STORE_KEY as storeKey } from '@woocommerce/block-data';
+import { useSelect } from '@wordpress/data';
+import { Cart } from '@woocommerce/type-defs/cart';
+import { SelectShippingRateType } from '@woocommerce/type-defs/shipping';
+import { useEffect, useRef } from '@wordpress/element';
+import { deriveSelectedShippingRates } from '@woocommerce/base-utils';
+import isShallowEqual from '@wordpress/is-shallow-equal';
+import { isObject } from '@woocommerce/types';
+
+/**
+ * Internal dependencies
+ */
+import { useSelectShippingRate } from './use-select-shipping-rate';
+
+type ShippingData = Pick<
```

Comment:
Is `Pick` prefered over redeclaring the props and/or importing the typedef per prop via `typeof`? e.g.

```
shippingRates: typeof Cart['shippingRates'];
```

Asking because I've done it the other way but not considered Pick.

Reply:
I'm not sure if it's preferred, but I think your way is actually a bit tidier, imo it makes it easier to read.